tests/queries.py

```diff
@@ -230,57 +227,47 @@ def parseAtx(log_messages):
     return node2blocks
 
 
+class Node:
+    def __init__(self):
+        self.blocks = 0
+        self.layers = defaultdict(int)
+
+    def add_block_to_layer(self, layer: int):
+        self.blocks += 1
+        self.layers[layer] += 1
+
+
 # sets log messages into a dictionary where keys=node_id and
 # value is a dictionary of blocks and layers
 # TODO this can be a util function
-def sort_by_nodeid(log_messages):
-    node2blocks = {}
+def sort_by_node_id(log_messages):
+    node2blocks = defaultdict(Node)
     for log in log_messages:
-        # I've created a block in layer %v. id: %v, num of transactions: %v, votes: %d,
-        # viewEdges: %d, atx %v, atxs:%v
-        id = re.split(r'\.', log.N)[0]
-        m = re.findall(r'\w+\b', log.M)
-        # layer field
-        layer = m[7]
-        # blocks - list of all blocks, layers - map of blocks per layer
-        if id in node2blocks:
-            node2blocks[id]["blocks"].append(m)
-            if layer in node2blocks[id]["layers"]:
-                node2blocks[id]["layers"][layer].append(m)
-            else:
-                node2blocks[id]["layers"][layer] = [m]
-        else:
-            node2blocks[id] = {"blocks": [m], "layers": {m[7]: [m]}}
+        node2blocks[log.node_id].add_block_to_layer(log.layer_id)
```

Comment:
this will not store all kinds of data that has helped me debug runs such as block id etc...

Reply:
fixed